### PR TITLE
refactor(variables): prefix all variables with gds

### PR DESCRIPTION
**Pull request by oxyc**
This was just some quick sed + vim macro magic to get the changes but I havent at all looked if there were failed edge cases.

Deprecations file should not be included in the actual build and if we choose to provide it, it would be for sites to ease migrations.

### CODE REVIEW

**Comment by taromorimoto** [comment]
Ooh! Nice! Let's merge!

**Comment by oxyc** [comment]
### ⚠ BREAKING CHANGES

* **variables:** All variables have now been prefixed with `--gds-`. You can use the following commands to rename them in your theme

```bash
find -E . -type f -not -path "./node_modules/*" -not -path "./vendor/*" -regex ".*\.(js|css|scss|tsx|php)" -exec sed -i '' -E 's/--(color|spacing|body|heading|label|paragraph|tag|button|text-button|navigation|card|media-card|logo-grid-item|icon|accordion)/--gds-\1/g' {} +

find -E . -type f -not -path "./node_modules/*" -not -path "./vendor/*" -regex ".*\.(js|css|scss|tsx|php)" -exec sed -i '' -E 's/spacing-xxxs/spacing-3xs/g' {} +
find -E . -type f -not -path "./node_modules/*" -not -path "./vendor/*" -regex ".*\.(js|css|scss|tsx|php)" -exec sed -i '' -E 's/spacing-xxs/spacing-2xs/g' {} +
find -E . -type f -not -path "./node_modules/*" -not -path "./vendor/*" -regex ".*\.(js|css|scss|tsx|php)" -exec sed -i '' -E 's/spacing-xxl/spacing-2xl/g' {} +
find -E . -type f -not -path "./node_modules/*" -not -path "./vendor/*" -regex ".*\.(js|css|scss|tsx|php)" -exec sed -i '' -E 's/spacing-xxxl/spacing-3xl/g' {} +
```
* **variables:** The following variables have changed name:

```css
--heading-2xl-font-family: var(--heading-xxl-font-family);
--heading-2xl-font-weight: var(--heading-xxl-font-weight);
--heading-2xl-font-size: var(--heading-xxl-font-size);
--heading-2xl-line-height: var(--heading-xxl-line-height);
--heading-2xl-text-transform: var(--heading-xxl-text-transform);
--heading-2xl-letter-spacing: var(--heading-xxl-letter-spacing);
--heading-2xl-margin-top: var(--heading-xxl-margin-top);
--heading-2xl-margin-bottom: var(--heading-xxl-margin-bottom);

--spacing-3xs: var(--spacing-xxxs);
--spacing-2xs: var(--spacing-xxs);
--spacing-2xl: var(--spacing-xxl);
--spacing-3xl: var(--spacing-xxxl);
```
* **variables:** The following variables have been deprecated, please use the specific colors instead.

```css
--color-primary: var(--color-ui-03);
--color-secondary: var(--color-ui-04);
--border-color-light: #dddddd;
--background-color-primary: var(--color-ui-background-01);
--background-color-secondary: var(--color-ui-background-02);
--text-color-primary: var(--color-black);
```

**Comment by toffebjorkskog** [comment]
Nice!
Great with the string replacement snippet

File: src/components/gds-heading/_index.scss [hunk 1]
```diff
@@ -13,14 +13,14 @@
...
-  margin-bottom: var(--heading-xxl-margin-bottom);
+  font-family: var(--heading-2xl-font-family, var(--heading-font-family));
+  font-size: var(--heading-2xl-font-size);
+  font-weight: var(--heading-2xl-font-weight);
+  line-height: var(--heading-2xl-line-height);
+  text-transform: var(--heading-2xl-text-transform);
+  letter-spacing: var(--heading-2xl-letter-spacing);
+  margin-top: var(--heading-2xl-margin-top);
+  margin-bottom: var(--heading-2xl-margin-bottom);
 }
```
Comment (toffebjorkskog) [inline src/components/gds-heading/_index.scss:24]: makes sense

File: src/components/gds-accordion/gds-accordion.scss [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
-  background: var(--accordion-background, var(--color-white));
-  border: var(--accordion-border);
-  border-radius: var(--accordion-border-radius);
-  box-shadow: var(--accordion-box-shadow);
+  color: var(--gds-accordion-color, inherit);
+  background: var(--gds-accordion-background, var(--gds-color-white));
+  border: var(--gds-accordion-border);
+  border-radius: var(--gds-accordion-border-radius);
+  box-shadow: var(--gds-accordion-box-shadow);
   position: relative;
```
Comment (toffebjorkskog) [inline src/components/gds-accordion/gds-accordion.scss:9]: good with prefix

**Review by toffebjorkskog — APPROVED** [review]

**Pull request merged**